docs/source/getting_started.rst

```diff
@@ -320,35 +320,140 @@ syntax like so:
    my_global_dict = {}  # type: Dict[int, float]
 
-.. _stubs-intro:
+.. _pep526: https://www.python.org/dev/peps/pep-0526/
+
 
-Library stubs and typeshed
-**************************
+Types and classes
+*****************
 
-Mypy uses library *stubs* to type check code interacting with library
-modules, including the Python standard library. A library stub defines
-a skeleton of the public interface of the library, including classes,
-variables and functions, and their types. Mypy ships with stubs for
-the standard library from the `typeshed
-<https://github.com/python/typeshed>`_ project, which contains library
-stubs for the Python builtins, the standard library, and selected
-third-party packages.
+So far, we've only seen examples of pre-existing types like the ``int``
+or ``float`` builtins, or the ``List`` and ``Dict`` types from the typing
+module. However, these aren't the only types you can use: in fact, you can
+use any Python class as a type!
 
-For example, consider this code:
+For example, suppose you've defined a custom class representing a bank account:
 
 .. code-block:: python
 
-  x = chr(4)
+    class BankAccount:
+        # Note: It is ok to omit type hints for the "self" parameter.
+        # Mypy will infer the correct type.
+
+        def __init__(self, account_name: str, initial_balance: int = 0) -> None:
+            # Note: Mypy will infer the correct types of your fields
+            # based on the types of the parameters.
+            self.account_name = account_name
+            self.balance = initial_balance
+
+        def deposit(self, amount: int) -> None:
+            self.balance += amount
+
+        def withdraw(self, amount: int) -> None:
+            self.balance -= amount
+
+        def overdrawn(self) -> bool:
+            return self.balance < 0
+
+You can declare that a function will accept any instance of your class
+by simply annotating the parameters with ``BankAccount``:
+
+.. code-block:: python
+
+    def transfer(src: BankAccount, dst: BankAccount, amount: int) -> None:
+        src.withdraw(amount)
+        dst.deposit(amount)
+
+    account_1 = BankAccount('Alice', 400)
+    account_2 = BankAccount('Bob', 200)
+    transfer(account_1, account_2, 50)
+
+In fact, the ``transfer`` function we wrote above can accept more then just
+instances of ``BankAccount``: it can also accept any instance of a *subclass*
+of ``BankAccount``. For example, suppose you write a new class that looks like this:
+
+.. code-block:: python
+
+    class AuditedBankAccount(BankAccount):
+        def __init__(self, account_name: str, initial_balance: int = 0) -> None:
+            super().__init__(account_name, initial_balance)
+
+            # In this case, mypy can't infer the exact type of this field
+            # based on the information available in this constructor, so we
+            # need to add a type annotation.
+            self.audit_log: List[str] = []
```

Comment:
I'm not massively keen on an example that requires a three-line comment, especially since this is a beginner's tutorial. Can we think of a different example here, that doesn't involve having to instantiate an empty list? (I'm also trying to think of one.)

In any event, we should use PEP 585 syntax here :)

```suggestion
            # In this case, mypy can't infer the exact type of this field
            # based on the information available in this constructor, so we
            # need to add a type annotation.
            self.audit_log: list[str] = []
```

Reply:
Maybe a transaction count as an int?

Reply:
I like that idea!

Reply:
IMHO it's not really a big deal since typing an empty list is a frequent case (personal feeling).

Reply:
mypy would be able to infer an int

Reply:
I'm going to keep the `list[str]` but remove the comments since the rule has been introduced in previous section